### DISCUSSION TITLE
Make debug and error messages more useful

### DISCUSSION THREAD

**Discussion by rjoleary** [u-root/u-root#2119] [General]
There are a number of debug/error prints which are missing key information like: which arguments? what's the error? etc...

**Comment by hugelgupf** [comment]
Can we expand on this? E.g.

Lots of commands in u-root have leftover "debug" functions that print stuff at various log levels. They often consist of messages like "it's here!" or "it's not here!" and are often only useful to the person who wrote them in the first place.

1. Maybe we need to standardize some kind of log package.

Yeah, I know, the entirety of the Go community is going to hate it. See #280, my first approach to that, which I somewhat abandoned. (Not for technical reasons, more for time investment reasons.) Maybe it should be revived.

2. Log messages need to say exactly what they're printing. E.g. "Received DHCP packet from X: %v" rather than just "%v"-splating the packet on the screen.

I think 2 can be expanded to a few more principles about logs (e.g. "don't print variable names", because that doesn't help users who have never looked at the source debug their issue), but I'm going to leave it like this.
